transport/internal/tlsmux/listener.go

```diff
@@ -21,45 +21,130 @@
 package tlsmux
 
 import (
+	"context"
 	"crypto/tls"
+	"errors"
 	"net"
+	"sync"
+	"time"
+
+	"go.uber.org/net/metrics"
+	"go.uber.org/zap"
 )
 
+var (
+	errListenerClosed = errors.New("listener closed")
+
+	_sniffReadTimeout    = time.Second * 10
+	_tlsHandshakeTimeout = time.Second * 10
+)
+
+// Config describes how listener should be configured.
+type Config struct {
+	Listener  net.Listener
+	TLSConfig *tls.Config
+
```

Comment:
nit: I don't think we need a new line here. Can you please remove this?

Reply:
Will address this in a separate PR